docs/api_guide/tutorial_pynapple_wavelets.py

```diff
@@ -0,0 +1,489 @@
+
+# %%
+# ***
+# Reconstructing the Slow Oscillation and Phase
+# ------------------
+# We can see that the decomposition has picked up on the 2Hz component of the signal, as well as the component with
+# increasing frequency. In this section, we will extract just the 2Hz component from the wavelet decomposition,
+# and see how it compares to the original section.
+
+# Get the index of the 2Hz frequency
+two_hz_freq_idx = np.where(freqs == 2.0)[0]
+# The 2Hz component is the real component of the wavelet decomposition at this index
+slow_oscillation = np.real(mwt[:, two_hz_freq_idx])
+# The 2Hz wavelet phase is the angle of the wavelet decomposition at this index
+slow_oscillation_phase = np.angle(mwt[:, two_hz_freq_idx])
+
+# %%
+# Lets plot it.
+fig = plt.figure(constrained_layout=True, figsize=(10, 4))
+axd = fig.subplot_mosaic(
+    [["signal"], ["phase"]],
+    height_ratios=[1, 0.4],
+)
+axd["signal"].plot(sig, label="Raw Signal", alpha=0.5)
+axd["signal"].plot(slow_oscillation, label="2Hz Reconstruction")
+axd["signal"].legend()
+axd["signal"].set_ylabel("Signal")
+
+axd["phase"].plot(slow_oscillation_phase, alpha=0.5)
+axd["phase"].set_ylabel("Phase (rad)")
+axd["phase"].set_xlabel("Time (s)")
+[axd[k].margins(0) for k in ["signal", "phase"]]
+
+# %%
+# ***
+# Adding in the 15Hz Oscillation
+# ------------------
+# Let's see what happens if we also add the 15 Hz component of the wavelet decomposition to the reconstruction. We
+# will extract the 15 Hz components, and also the 15Hz wavelet power over time. The wavelet power tells us to what
+# extent the 15 Hz frequency is present in our signal at different times.
+#
+# Finally, we will add this 15 Hz reconstruction to the one shown above, to see if it improves out reconstructed
+# signal.
+
+# Get the index of the 15 Hz frequency
+fifteen_hz_freq_idx = np.where(freqs == 15.0)[0]
+# The 15 Hz component is the real component of the wavelet decomposition at this index
+fifteenHz_oscillation = np.real(mwt[:, fifteen_hz_freq_idx])
+# The 15 Hz poser is the absolute value of the wavelet decomposition at this index
+fifteenHz_oscillation_power = np.abs(mwt[:, fifteen_hz_freq_idx])
+
+# %%
+# Lets plot it.
+
+fig = plt.figure(constrained_layout=True, figsize=(10, 4))
+gs = plt.GridSpec(2, 1, figure=fig, height_ratios=[1.0, 1.0])
+
+ax0 = plt.subplot(gs[0, 0])
+ax0.plot(fifteenHz_oscillation, label="15Hz Reconstruction")
+ax0.plot(fifteenHz_oscillation_power, label="15Hz Power")
+ax0.set_xticklabels([])
+
+ax1 = plt.subplot(gs[1, 0])
+ax1.plot(sig, label="Raw Signal", alpha=0.5)
+ax1.plot(slow_oscillation + fifteenHz_oscillation.values, label="2Hz + 15Hz Reconstruction")
+ax1.set_xlabel("Time (s)")
+
+[
+    (a.margins(0), a.legend(), a.set_ylim(-2.5, 2.5), a.set_ylabel("Signal"))
+    for a in [ax0, ax1]
+]
+
+
+# %%
+# ***
+# Adding ALL the Oscillations!
+# ------------------
+# Let's now add together the real components of all frequency bands to recreate a version of the original signal.
+
```

Comment:
maybe it is worth anticipating what is going to happen . We will now learn how to interpret the parameters of the wavelet, and in particular how to trade off the accuracy in the frequency decomposition with the accuracy in the time domain reconstruction; 

basically introducing the concepts before showing the initial failure. I think it makes for a better flow